### ISSUE TITLE
Webpack angular usage

### ISSUE THREAD

**Issue by lukashlobil** [NathanaelA/nativescript-localstorage#3]
How do I use this plugin webpack? I used it w/o webpack and everything worked as expected. I have the webpack-ready version downloaded, but am getting:
```
ERROR Error: Uncaught (in promise): ReferenceError: localStorage is not defined
ReferenceError: localStorage is not defined
     at MyComponent.module.exports.Array.concat.MyComponent.ngOnInit (file:///data/data/cz.datasys.bavic.dev/files/app/bundle.js:52021:26)
     at checkAndUpdateDirectiveInline (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:11239:19)
     at checkAndUpdateNodeInline (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:12617:17)
     at checkAndUpdateNode (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:12585:16)
     at debugCheckAndUpdateNode (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:13214:59)
     at debugCheckDirectivesFn (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:13155:13)
     at Object.updateDirectives (file:///data/data/cz.datasys.bavic.dev/files/app/bundle.js:64540:9)
     at Object.debugUpdateDirectives [as updateDirectives] (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:13140:21)
     at checkAndUpdateView (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:12552:14)
     at callWithDebugContext (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:13540:42)
     at Object.debugCheckAndUpdateView [as checkAndUpdateView] (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:13080:12)
     at ViewRef_.module.exports.ViewRef_.detectChanges (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:10650:63)
     at RouterOutlet.module.exports.RouterOutlet.activateWith (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:22621:42)
     at ActivateRoutes.module.exports.ActivateRoutes.placeComponentIntoOutlet (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:21799:16)
     at ActivateRoutes.module.exports.ActivateRoutes.activateRoutes (file:///data/data/cz.datasys.bavic.dev/files/app/vendor.js:21780:26)
```
Do I have to define it some other way? I played around with requiring it in a component, wrapping it in a service, still getting the same error though.

**Comment by dakotamurphyucf** [comment]
@lukashlobil  if you remove the ".js" extention in  const FileSystemAccess = require('file-system/file-system-access.js').FileSystemAccess; in the node_module it will work

**Comment by DanielKucal** [comment]
I also have issues with this plugin... I use webpack configuration as in `sample-Groceries`. Unchanged version fails on compilation, so I've tried  @dakotamurphyucf's solution and @sis0k0's fork, but then I get runtime error shown below. I was trying wih `require('nativescript-localstorage')` in different places in `main.ts`, `main.aot.ts`, `vendor.ts` and directly in the files using localstorage, but it fails everytime:
```
CONSOLE ERROR file:///app/vendor.js:2164:24: ERROR Error: Uncaught (in promise): Error: Expecting function got: undefined
	wrap@file:///app/vendor.js:58933:32 [angular]
	zonedCallback@file:///app/vendor.js:49191:40 [angular]
	setTimeout@file:///app/vendor.js:63033:45 [angular]
	setTimeout [angular]
	show@file:///app/bundle.js:6861:42 [angular]
	responseInterceptor@file:///app/bundle.js:32787:36 [angular]
	value@file:///app/bundle.js:34676:58 [angular]
	testConnection@file:///app/bundle.js:26896:18 [angular]
	LoginComponent@file:///app/bundle.js:32497:33 [angular]
	LoginComponent@[native code] [angular]
	createClass@file:///app/vendor.js:11881:81 [angular]
	createDirectiveInstance@file:///app/vendor.js:11707:48 [angular]
	createViewNodes@file:///app/vendor.js:13057:72 [angular]
	createRootView@file:///app/vendor.js:12962:20 [angular]
	callWithDebugContext@file:///app/vendor.js:14093:47 [angular]
	create@file:///app/vendor.js:10898:60 [angular]
	createComponent@file:///app/vendor.js:11098:68 [angular]
	activateOnGoForward@file:///app/vendor.js:48419:70 [angular]
	activateWith@file:///app/vendor.js:48405:37 [angular]
	placeComponentIntoOutlet@file:///app/vendor.js:23191:28 [angular]
	activateRoutes@file:///app/vendor.js:23172:50 [angular]
	file:///app/vendor.js:23108:72 [angular]
	forEach@[native code] [angular]
	activateChildRoutes@file:///app/vendor.js:23108:36 [angular]
	activate@file:///app/vendor.js:23082:33 [angular]
	file:///app/vendor.js:22699:30 [angular]
	file:///app/vendor.js:230:25 [angular]
	__tryOrSetError@file:///app/vendor.js:641:20 [angular]
	next@file:///app/vendor.js:583:42 [angular]
	_next@file:///app/vendor.js:523:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyNext@file:///app/vendor.js:52414:34 [angular]
	_next@file:///app/vendor.js:68701:31 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_subscribe@file:///app/vendor.js:52130:28 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	_innerSub@file:///app/vendor.js:52401:55 [angular]
	_tryNext@file:///app/vendor.js:52398:23 [angular]
	_next@file:///app/vendor.js:52381:26 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyNext@file:///app/vendor.js:52414:34 [angular]
	_next@file:///app/vendor.js:68701:31 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_complete@file:///app/vendor.js:48826:34 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_complete@file:///app/vendor.js:52406:38 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_subscribe@file:///app/vendor.js:39590:32 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	_innerSub@file:///app/vendor.js:52401:55 [angular]
	_tryNext@file:///app/vendor.js:52398:23 [angular]
	_next@file:///app/vendor.js:52381:26 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyNext@file:///app/vendor.js:52414:34 [angular]
	_next@file:///app/vendor.js:68701:31 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyNext@file:///app/vendor.js:52414:34 [angular]
	_next@file:///app/vendor.js:68701:31 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyComplete@file:///app/vendor.js:61573:30 [angular]
	_complete@file:///app/vendor.js:61590:28 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_complete@file:///app/vendor.js:52406:38 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_subscribe@file:///app/vendor.js:39590:32 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	_innerSub@file:///app/vendor.js:52401:55 [angular]
	_tryNext@file:///app/vendor.js:52398:23 [angular]
	_next@file:///app/vendor.js:52381:26 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyComplete@file:///app/vendor.js:61573:30 [angular]
	_complete@file:///app/vendor.js:61590:28 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_complete@file:///app/vendor.js:52406:38 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_subscribe@file:///app/vendor.js:39590:32 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	_innerSub@file:///app/vendor.js:52401:55 [angular]
	_tryNext@file:///app/vendor.js:52398:23 [angular]
	_next@file:///app/vendor.js:52381:26 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyNext@file:///app/vendor.js:52414:34 [angular]
	_next@file:///app/vendor.js:68701:31 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_subscribe@file:///app/vendor.js:52130:28 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	_innerSub@file:///app/vendor.js:52401:55 [angular]
	_tryNext@file:///app/vendor.js:52398:23 [angular]
	_next@file:///app/vendor.js:52381:26 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	notifyNext@file:///app/vendor.js:52414:34 [angular]
	_next@file:///app/vendor.js:68701:31 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_subscribe@file:///app/vendor.js:52130:28 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	_innerSub@file:///app/vendor.js:52401:55 [angular]
	_tryNext@file:///app/vendor.js:52398:23 [angular]
	_next@file:///app/vendor.js:52381:26 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:523:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_next@file:///app/vendor.js:42408:30 [angular]
	next@file:///app/vendor.js:487:23 [angular]
	_complete@file:///app/vendor.js:61879:29 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_complete@file:///app/vendor.js:42517:38 [angular]
	complete@file:///app/vendor.js:512:27 [angular]
	_subscribe@file:///app/vendor.js:52132:36 [angular]
	_trySubscribe@file:///app/vendor.js:180:35 [angular]
	subscribe@file:///app/vendor.js:168:40 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	subscribe@file:///app/vendor.js:165:26 [angular]
	file:///app/vendor.js:209:47 [angular]
```
Any ideas or alternatives?

**Comment by NathanaelA** [comment]
@dakotamurphyucf is correct; this is a requirement for the new version of webpack; in my next release I'll fix it...

**Comment by NathanaelA** [comment]
I have released v1.1.1 that fixes webpack issues.

**Comment by lukashlobil** [comment]
The .js extension is back in 1.1.1 and that doesn't work with webpack

**Comment by NathanaelA** [comment]
@lukashlobil - I'm not sure what you are talking about...  No, the `.js` extension is not...   

Here is the github commit; you can see it has no .js extension
https://github.com/NathanaelA/nativescript-localstorage/blob/master/package.json#L5

Here is a fresh install from NPM of nativescript-localstorage @ 1.1.1  (See bottom of screen shot for `version`, top of screen shot to the `main` key)
![image](https://cloud.githubusercontent.com/assets/850871/25316203/e861d07a-2827-11e7-9d57-e418ea5641c4.png)

**Comment by DanielKucal** [comment]
@NathanaelA, I'm sure he means this one. Screenshot is from sis0k0's pull request.
![image](https://cloud.githubusercontent.com/assets/8938562/25316230/1872a7ea-2863-11e7-833c-6253eeb7be16.png)

**Comment by NathanaelA** [comment]
@dakotamurphyucf - Ah, very good catch;  guess I just can't read I just noticed someone mentioned that above.  Just published v1.1.2 which fixes that.   Thanks!

**Comment by EddyMurphy95** [comment]
hello, I am trying to  use local storage in nativescript and this is what i am getting.
ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'getItem' of undefined
